source/sundials/kinsol.cc

```diff
@@ -451,6 +451,18 @@ namespace SUNDIALS
         AssertKINSOL(status);
       }
 
+    // Right before calling the main KINSol() call, allow expert users to
+    // perform additional setup operations on the KINSOL object.
+    if (custom_setup)
+      {
+        status =
+          Utilities::call_and_possibly_capture_exception(custom_setup,
```

Comment:
@bangerth Do you know if I need this here?

Reply:
No. You haven't entered KINSOL code here yet, and so if `custom_setup` issues an exception, it will simply propagate through our own wrapper code and back into the calling code. The exception capture stuff is only necessary if we have to propagate exceptions through KINSOL.